django/contrib/auth/base_user.py

```diff
@@ -21,6 +23,10 @@ def normalize_email(cls, email):
         """
         Normalize the email address by lowercasing the domain part of it.
         """
+        msg = 'BaseUserManager.normalize_email() is deprecated in favor of '
+        msg += 'AbstractUser.normalize_email() and will be removed in Django 4.1.'
+        warnings.warn(msg, RemovedInDjango41Warning, stacklevel=2)
+
         email = email or ''
```

Comment:
To avoid duplication, this could call the method in its new location with `AbstractUser.normalize_email`.

Reply:
Now that I've rewritten this a bit, I don't see a way to raise this warning in just one place. I think I need to raise it each time I detect an overridden method. If I'm missing an opportunity to consolidate, let me know.

Reply:
Oh got it, this is the same suggestion as @smithdc1's below. I thought you were referring to the raising of the warning specifically, not the functionality. I explored this at first and ran into circular import issues but this is possible again now that we've moved it. Thanks.